software/contrib/sequential_switch.py

```diff
@@ -0,0 +1,319 @@
+                    if self.parent.current_output == out_no:
+                        switches = switches + " [*] "
+                    else:
+                        switches = switches + " [ ] "
+                else:
+                    # output is disabled; mark it with .
+                    switches = switches + "  .  "
+                    
+            switches = switches + "\n"
+        
+        oled.centre_text(switches)
+        
+        oled.show()
+
+class MenuScreen:
+    """Advanced menu options screen
+    """
+    def __init__(self, parent):
+        self.parent = parent
+        
+        self.menu_items = [
+            NumOutsChooser(parent),
+            ModeChooser(parent)
+        ]
+        
+    def draw(self):
+        self.menu_items[self.parent.menu_item].draw()
+        
+    def on_button1(self):
+        self.menu_items[self.parent.menu_item].on_button1()
+
+class NumOutsChooser:
+    """Used by MenuScreen to choose the number of outputs
+    """
+    
+    def __init__(self, parent):
+        self.parent = parent
+        
+    def read_num_outs(self):
+        return round(knob_rescale(k2, 2, 6)) # 2-6 outputs; 1 output would be useless!
+        
+    def on_button1(self):
+        num_outs = self.read_num_outs()
+        self.parent.num_outputs = num_outs
+        self.parent.current_output = self.parent.current_output % num_outs
+        self.parent.save()
+        
+    def draw(self):
+        num_outs = self.read_num_outs()
+        oled.fill(0)
+        oled.text(f"-- # Outputs --", 0, 0)
+        oled.text(f"{self.parent.num_outputs} <- {num_outs}", 0, 10)
+        oled.show()
+    
+class ModeChooser:
+    """Used by MenuScreen to choose the operating mode
+    """
+    def __init__(self, parent):
+        self.parent = parent
+        
+        self.mode_names = [
+            "Seq.",
+            "Rev.",
+            "P-P",
+            "Rand."
+        ]
+        
+    def read_mode(self):
+        return round(knob_rescale(k2, 0, len(self.mode_names)-1))
+        
+    def on_button1(self):
+        new_mode = self.read_mode()
+        self.parent.mode = new_mode
+        self.parent.save()
+        
+    def draw(self):
+        new_mode = self.read_mode()
+        oled.fill(0)
+        oled.text(f"-- Mode --", 0, 0)
+        oled.text(f"{self.mode_names[self.parent.mode]} <- {self.mode_names[new_mode]}", 0, 10)
+        oled.show()
+
+class SequentialSwitch(EuroPiScript):
+    """The main workhorse of the whole module
+
+    Copies the analog input to one of the 6 outputs, cycling which output
+    whenever a trigger is received
+    """
+    
+    def __init__(self):
+        super().__init__()
+        
+        # keep track of the last time the user interacted with the module
+        # if we're idle for too long, start the screensaver
+        self.last_interaction_time = time.ticks_ms()
+        
+        # Continious quantizing, or only on an external trigger?
```

Comment:
[optional] This looks like a copied comment from the quantizer code.

Reply:
Whoops. Yes, the dangers of re-using code without reading carefully.